src/basics/custom.md

```diff
@@ -0,0 +1,493 @@
+impl Associated for CustomContract {
+    type Error = StdError;
+    type ExecC = ExternalMsg;
+    type QueryC = ExternalQuery;
+
+    #[msg(exec)]
+    fn associated_exec(
+        &self,
+        _ctx: ExecCtx<Self::QueryC>,
+    ) -> Result<Response<Self::ExecC>, Self::Error> {
+        Ok(Response::new())
+    }
+
+    #[msg(query)]
+    fn associated_query(&self, _ctx: QueryCtx<Self::QueryC>) -> Result<String, Self::Error> {
+        Ok(String::default())
+    }
+}
+```
+
+## `messages` attribute on main `contract` call
+
+We implemented the custom interfaces on the contract. Now, we can use it in the contract itself.
+Because the contract is already defined as `custom,` we only need to use the `messages` attribute as in the case of
+non-custom interfaces.
+
+```rust
+use cosmwasm_std::{Response, StdResult};
+use sylvia::{contract, types::InstantiateCtx};
+
+use crate::external_messages::{ExternalMsg, ExternalQuery};
+
+pub struct CustomContract;
+
+#[contract]
+#[messages(crate::sv_custom as SvCustomInterface)]
+#[messages(crate::associated as AssociatedInterface)]
+#[sv::custom(msg=ExternalMsg, query=ExternalQuery)]
+impl CustomContract {
+    pub const fn new() -> Self {
+        Self
+    }
+
+    #[msg(instantiate)]
+    pub fn instantiate(
+        &self,
+        _ctx: InstantiateCtx<ExternalQuery>,
+    ) -> StdResult<Response<ExternalMsg>> {
+        Ok(Response::new())
+    }
+}
+```
+
+Nice and easy, isn't it?
+Only one thing that needs to be added. I mentioned earlier that it is possible to implement a non-custom interface on a custom contract.
+Let's define a non-custom interface.
+
+```rust
+use cosmwasm_std::{Response, StdError};
+use sylvia::interface;
+use sylvia::types::{ExecCtx, QueryCtx};
+
+#[interface]
+pub trait NonCustom {
+    type Error: From<StdError>;
+
+    #[msg(exec)]
+    fn non_custom_exec(&self, ctx: ExecCtx) -> Result<Response, Self::Error>;
+
+    #[msg(query)]
+    fn non_custom_query(&self, ctx: QueryCtx) -> Result<String, Self::Error>;
+}
+
+pub mod impl_non_custom {
+    use crate::contract::CustomContract;
+    use cosmwasm_std::{Response, StdError};
+    use sylvia::contract;
+    use sylvia::types::{ExecCtx, QueryCtx};
+
+    use super::NonCustom;
+
+    #[contract(module=crate::contract)]
+    #[messages(crate::non_custom as NonCustom)]
+    #[sv::custom(msg=ExternalMsg, query=ExternalQuery)]
+    impl NonCustom for CustomContract {
+        type Error = StdError;
+
+        #[msg(exec)]
+        fn non_custom_exec(&self, _ctx: ExecCtx) -> Result<Response, Self::Error> {
+            Ok(Response::new())
+        }
+
+        #[msg(query)]
+        fn non_custom_query(&self, _ctx: QueryCtx) -> Result<String, Self::Error> {
+            Ok(String::default())
+        }
+    }
+}
+```
+
+As you can see, although it's non-custom, we still have to inform `sylvia` custom types from the contract.
+It's required for the multitest helpers to be generic over the same types as the contract.
```

Comment:
We have agreed with product and marketing team, that the official name is **MultiTest**.
So please just replace multitest with MultiTest ;-)